qt/python/mantidqt/project/test/test_project.py

```diff
@@ -97,7 +97,7 @@ def test_offer_save_does_nothing_if_saved_is_true(self):
 
     def test_offer_save_does_something_if_saved_is_false(self):
         self.project._offer_save_message_box = mock.MagicMock(return_value=QMessageBox.Yes)
-        self.project.save = mock.MagicMock()
+        self.project.save = mock.MagicMock(return_value=None)
```

Comment:
Is a new test required where the return value here is True?

Reply:
I'll add to this PR with a new unit test and a fix so that closing the save as dialog using the X doesn't close Workbench.